packages/core/src/messagingProtocols/proteus/ProteusService/DecryptionErrorGenerator/DecryptionErrorGenerator.ts

```diff
@@ -18,46 +18,60 @@
  */
 
 import {QualifiedId} from '@wireapp/api-client/lib/user';
-import logdown from 'logdown';
 
 import {DecryptionError} from '../../../../errors/DecryptionError';
 
 type ErrorWithCode = Error & {code?: number};
 const hasErrorCode = (error: any): ErrorWithCode => error && error.code;
 
-const generateDecryptionError = (
-  senderInfo: {clientId: string; userId: QualifiedId},
-  error: any,
-  logger: logdown.Logger,
-): DecryptionError => {
-  const errorCode = hasErrorCode(error) ? error.code ?? 999 : 999;
-  let message = 'Unknown decryption error';
+export const ProteusErrors = {
+  InvalidMessage: 201,
+  RemoteIdentityChanged: 204,
+  InvalidSignature: 207,
+  Unknown: 999,
+} as const;
 
-  const {clientId: remoteClientId, userId} = senderInfo;
-  const {id: remoteUserId} = userId;
+type CoreCryptoErrors = keyof typeof ProteusErrors;
+type ProteusErrorCode = typeof ProteusErrors[CoreCryptoErrors];
 
-  const isDuplicateMessage = false; // TODO when coreCrypto returns ganular errors
-  const isOutdatedMessage = false; // TODO when coreCrypto returns ganular errors
-  // We don't need to show these message errors to the user
-  if (isDuplicateMessage || isOutdatedMessage) {
-    message = `Message from user ID "${remoteUserId}" will not be handled because it is outdated or a duplicate.`;
-  }
+const CoreCryptoErrorMapping: Record<CoreCryptoErrors, ProteusErrorCode> = {
+  InvalidMessage: ProteusErrors.InvalidMessage,
+  RemoteIdentityChanged: ProteusErrors.RemoteIdentityChanged,
+  InvalidSignature: ProteusErrors.InvalidSignature,
+  Unknown: ProteusErrors.Unknown,
+};
 
-  const isInvalidMessage = false; // TODO when coreCrypto returns ganular errors
-  const isInvalidSignature = false; // TODO when coreCrypto returns ganular errors
-  const isRemoteIdentityChanged = false; // TODO when coreCrypto returns ganular errors
-  // Session is broken, let's see what's really causing it...
-  if (isInvalidMessage || isInvalidSignature) {
-    message = `Session with user '${remoteUserId}' (${remoteClientId}) is broken.\nReset the session for possible fix.`;
-  } else if (isRemoteIdentityChanged) {
-    message = `Remote identity of client '${remoteClientId}' from user '${remoteUserId}' changed`;
-  }
+const mapCoreCryptoError = (error: any): ProteusErrorCode => {
+  const code = hasErrorCode(error) ? error.code : CoreCryptoErrorMapping[error.message as CoreCryptoErrors];
```

Comment:
ie: this seems more correct (referencing earlier comment)
```suggestion
  const code = hasErrorCode(error) ?? CoreCryptoErrorMapping[error.message as CoreCryptoErrors];
```

Reply:
i personally prefer it returning a boolean but if we would like to keep the current function i think this is nicer. (i would rename the function in that case however)